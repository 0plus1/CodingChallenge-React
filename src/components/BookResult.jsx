import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardMedia, CardContent, Avatar, Typography } from '@material-ui/core';
import styled from 'styled-components';

const StyledCard = styled(Card)`
  min-width: 288px;
`;

const StyledCardMedia = styled(CardMedia)`
  height: 200px;
`;

const StyledAuthor = styled.p``;
const StyledISBN = styled.p``;

const BookResult = ({ name, isbn, publishedAt, cover, author }) => (
  <StyledCard aria-label={name}>
    <CardHeader
      avatar={<Avatar aria-label="book">B</Avatar>}
      title={name}
      subheader={publishedAt}
    />
    <StyledCardMedia
      image={cover}
      title={name}
    />
    <CardContent>
      <Typography variant="body2" color="textSecondary" component="div">
        <StyledAuthor><strong>Author</strong> {author}</StyledAuthor>
        <StyledISBN><strong>ISBN</strong> {isbn}</StyledISBN>
      </Typography>
    </CardContent>
  </StyledCard>
);

BookResult.propTypes = {
  name: PropTypes.string.isRequired,
  isbn: PropTypes.string,
  publishedAt: PropTypes.string,
  cover: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
};

BookResult.defaultProps = {
  isbn: null,
  publishedAt: null,
};

export default BookResult;
