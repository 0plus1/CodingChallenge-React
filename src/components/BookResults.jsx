import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';
import { BookResult } from './index';

const BookResults = ({ books }) => (
  <Grid container direction={'row'} justify={'center'} alignItems={'center'} spacing={4}>
    {books.map(book =>
      <Grid item xs={12} sm={6} md={4} lg={4} xl={3} key={`book-grid-${book.bookId}`}>
        <BookResult key={`book-${book.bookId}`} {...book} />
      </Grid>
    )}
  </Grid>
);

BookResults.propTypes = {
  books: PropTypes.arrayOf(PropTypes.shape(BookResult.propTypes)).isRequired,
};

export default BookResults;
