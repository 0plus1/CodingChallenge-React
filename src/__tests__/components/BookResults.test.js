import React from 'react';
import { shallow, mount } from 'enzyme';
import BookResults from '../../components/BookResults';
import BookResult from '../../components/BookResult';

const books = [
  {
    bookId: 1,
    name: 'Becker West Arnoldo',
    isbn: '3181444340',
    publishedAt: '2000-01-01',
    author: 'Mrs. John Doe',
    cover: 'https://lorempixel.com/640/480/?82539',
  },
  {
    bookId: 2,
    name: 'Schuster Champlinborough',
    isbn: '1441614311',
    publishedAt: '1991-07-29',
    author: 'Prof. John Doe',
    cover: 'https://lorempixel.com/640/480/?84662',
  },
];

describe('<BookResults />', () => {
  it('renders without crashing', () => {
    shallow(<BookResults books={books} />);

    expect(<BookResults books={books}/>).toMatchSnapshot();
  });

  it('renders with correct n children', () => {
    const wrapper = mount(<BookResults books={books} />);
    const bookChild = wrapper.find(BookResult);
    expect(bookChild.length)
      .toEqual((books.length));
    bookChild.forEach((child, index) => {
      expect(child.props())
        .toMatchObject(books[index]);
    });
    wrapper.unmount();
  });
});
