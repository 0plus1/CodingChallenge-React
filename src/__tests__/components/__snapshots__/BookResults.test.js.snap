// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookResults /> renders without crashing 1`] = `
<BookResults
  books={
    Array [
      Object {
        "author": "Mrs. John Doe",
        "bookId": 1,
        "cover": "https://lorempixel.com/640/480/?82539",
        "isbn": "3181444340",
        "name": "Becker West Arnoldo",
        "publishedAt": "2000-01-01",
      },
      Object {
        "author": "Prof. John Doe",
        "bookId": 2,
        "cover": "https://lorempixel.com/640/480/?84662",
        "isbn": "1441614311",
        "name": "Schuster Champlinborough",
        "publishedAt": "1991-07-29",
      },
    ]
  }
/>
`;
